### PR TITLE
Support wrapped token and native token in one implementation

### PR DESCRIPTION
**Pull request by bun919tw**

### CODE REVIEW

**Comment by bun919tw** [comment]
1. All functions of mint / borrow / redeem / redeemUnderlying / repay / repayBehalf / liquidateBorrow / _addReserves have a mirror functions in native token. Don't change the existing public function signatures since crWBNB (wrapped native delegator) on BSC has already been deployed.
2. Keep native token as cash (liquidity).
3. Can only flashloan and reducing reserves in the native token.

**Comment by bun919tw** [comment]
Note: The only cream wrapped native contract aiming to integrate the native token is crWBNB (on BSC). It will need to consider backward compatibility. For example, the original contract has `internalCash` but this storage is no longer needed. We will handle this by submitting another PR only to bsc branch after this one.

* Still need to include `internalCash` in storage but clear its value when `_becomeImplementation`.
* Unwrap the existing cash (liquidity) to the native token.
* Check if `gulp` function being undefined is ok or not.

File: contracts/CWrappedNative.sol [hunk 1]
```diff
@@ -31,103 +40,252 @@ contract CWrappedNative is CToken, CErc20Interface {
...
     /**
      * @notice Sender redeems cTokens in exchange for the underlying asset
      * @dev Accrues interest whether or not the operation succeeds, unless reverted
+     *  Keep return in the function signature for backward compatibility
      * @param redeemTokens The number of cTokens to redeem into underlying
      * @return uint 0=success, otherwise a failure (see ErrorReporter.sol for details)
      */
     function redeem(uint redeemTokens) external returns (uint) {
-        return redeemInternal(redeemTokens);
+        require(redeemInternal(redeemTokens, false) == 0, "redeem failed");
```
Comment (jeremyHD) [inline contracts/CWrappedNative.sol:79]: no more return?

Comment (jeremyHD) [reply contracts/CWrappedNative.sol:79]: oh I see, replace graceful failure with require

Comment (bun919tw) [reply contracts/CWrappedNative.sol:79]: Yes. Return is not important but keeps it for function signature consistency.

**Review by jeremyHD — APPROVED** [review]

**Comment by bun919tw** [comment]
Note 2: The function signature of `flashLoan` is not consistent between CWrappedNative and the others (one receiver is payable and others are not). However, we will refactor the flash loan implementation to follow EIP-3156 after this PR. Will also align the function signature at that moment.

**Review by jeremyHD — APPROVED** [review]

**Pull request merged**